homeassistant/components/danfoss_air/binary_sensor.py

```diff
@@ -0,0 +1,55 @@
+"""
+Support for the for Danfoss Air HRV binary sensor platform.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.danfoss_air/
+"""
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.danfoss_air import DOMAIN
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the available Danfoss Air sensors etc."""
+    from pydanfossair.commands import ReadCommand
+    data = hass.data[DOMAIN]
+
+    sensors = [["Danfoss Air Bypass Active", ReadCommand.bypass]]
+
+    dev = []
+
+    for sensor in sensors:
+        dev.append(DanfossAirBinarySensor(data, sensor[0], sensor[1]))
+
+    add_devices(dev, True)
+
+
+class DanfossAirBinarySensor(BinarySensorDevice):
+    """Representation of a Danfoss Air binary sensor."""
+
+    def __init__(self, data, name, sensor_type):
+        """Initialize the Danfoss Air binary sensor."""
+        self._data = data
+        self._name = name
+        self._state = None
+        self._type = sensor_type
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Type of device class."""
+        return "opening"
+
+    def update(self):
+        """Fetch new state data for the sensor."""
+        self._data.update()
```

Comment:
Since this data object is shared between all sensors and binary sensors of the component, if each one is calling update, isn't that going to cause race conditions on what's getting stored in the object?

Why not just schedule those updates once and then when each sensor/binary sensor calls update, it refreshes from the last update of data?

Reply:
I don't think so. The @Throtte annotation ensures that the method is not executed more often than the throttle interval. At least this is what the logs say. The idea was stolen from another component. 

If there is a better way to schedule the call of the update method please point me at some documentation or example.

Reply:
I missed the throttle, this should be fine then.